grama/eval_pnd.py

```diff
@@ -0,0 +1,346 @@
+all = [
+    "eval_pnd",
+    "ev_pnd",
+    "pareto_min_rel",
+    "make_proposal_sigma",
+    "rprop",
+    "dprop",
+    "approx_pnd"
+]
+
+from grama import add_pipe, Model, tf_md
+
+from numpy import array, diag, dot, ones, mean, zeros
+from numpy import any as npany
+from numpy import all as npall
+from numpy import min as npmin
+from numpy import max as npmax
+from numpy import sum as npsum
+from numpy.linalg import norm
+from numpy.random import choice, multivariate_normal
+from numpy.random import seed as set_seed
+from pandas import DataFrame
+from scipy.linalg import svd
+from scipy.stats import multivariate_normal as mvnorm
+from toolz import curry
+
+
+@curry
+def eval_pnd(model, df_train, df_test, sign, n=int(1e4), seed=None):
+    """ Evaluate a Model using a predictive model
+
+    Evaluates a given model against a PND algorithm to determine
+    "optimal points".
+
+    Args:
+        model (gr.model): predictive model to evaluate
+        df_train (DataFrame): dataframe with training data
+        df_test (DataFrame): dataframe with test data
+        sign (numpy array of +/-1 values): Array of optimization signs: {-1: Minimize, +1 Maximize}
```

Comment:
This should be a `dict`, so that the user can down-select to only the outputs they want to consider. For instance, a model may have `out = ["y0_mean", "y1_mean", "y2_mean"]`, but they only want to use `y1` and `y2` for PND. Then the call would be `sign = dict(y1_mean=+1, y2_mean=-1)`.